### DISCUSSION TITLE
Loki configuration validation on startup e.g. URL protocol

### DISCUSSION THREAD

**Discussion by OtegaOvie** [serilog-contrib/serilog-sinks-grafana-loki#163] [Ideas]
So, I spent over 6hrs troubleshooting why my logs are not sent to Loki as they will not show up on Grafana as Datasource. I did all I can, tweaked configuration and many other changes. I had to pull the project code and attached the project as a dependency to my project.
This was when i noticed that the failures were due to not having a protocol before the URL such as "http" and this was a pain.

I think this an other parameters that could break the log process should be validated for semantic correctness when the application is launched.

**Comment by mishamyte** [comment]
Sounds pretty reasonable, but could be tricky because of primitives use + config from JSON. I think partially it is restricted with Serilog's config system.

But I think this should be a point of investigation
